tools/git2pantheon/main.go

```diff
@@ -33,20 +33,30 @@ func cloneBranch(w http.ResponseWriter, r *http.Request) {
 		var repo repository_and_branch
 		err = json.Unmarshal(body, &repo)
 
-		log.Println(repo.Repo)
-		log.Println(repo.Branch)
+		//check parsing error. If error is not null, log and return 400 if parse failed
+		if err != nil {
+			log.Printf("Error in unmarshalling request body %s", err)
+			//setting the response in JSON format for standardizing error responses
+			response := "{\"error\" : \"Error reading request body due to " + err.Error() + " \"}"
+			http.Error(w, response, http.StatusBadRequest)
+			return
+		}
+
+		log.Println("repo:  " + repo.Repo)
+		log.Println("branch: " + repo.Branch)
 
 		repository := repo.Repo
-		error_status := false
+
 		if repository == "" {
-			http.Error(w, "Error reading repository url", http.StatusInternalServerError)
-			error_status = true
+			//setting the response in JSON format for standardizing error responses
+			http.Error(w, "{\"error\" : \"Error reading repository url\"}", http.StatusInternalServerError)
+			// return after setting error response to avoid superfluous error header in logs
+			return
 		}
 		if !strings.Contains(strings.ToUpper(repository), "GIT") {
-			http.Error(w, "The repository entered does not look like a git repo.", http.StatusInternalServerError)
```

Comment:
I recall we currently only support git repo.

Reply:
I do not have context about why this check is there. The main focus of this PR is to add error handling, providing response for each error that is caught, and make the error response standard. Hence, I did not change the check. Anyways we are going to rewrite the git2pantheon later, we can make discuss it in a JIRA  related to that. Sounds good?